src/core/server/integration_tests/saved_objects/migrations/zdt_1/basic_downgrade.test.ts

```diff
@@ -0,0 +1,163 @@
+
+describe('ZDT upgrades - basic downgrade', () => {
+  let esServer: TestElasticsearchUtils['es'];
+
+  const startElasticsearch = async () => {
+    const { startES } = createTestServers({
+      adjustTimeout: (t: number) => jest.setTimeout(t),
+      settings: {
+        es: {
+          license: 'basic',
+        },
+      },
+    });
+    return await startES();
+  };
+
+  beforeAll(async () => {
+    await fs.unlink(logFilePath).catch(() => {});
+    esServer = await startElasticsearch();
+  });
+
+  afterAll(async () => {
+    await esServer?.stop();
+    await delay(10);
+  });
+
+  const typeV1 = createType({
+    name: 'sample_type',
+    mappings: {
+      properties: {
+        text: { type: 'text' },
+        keyword: { type: 'keyword' },
+      },
+    },
+    switchToModelVersionAt: '8.0.0',
+    modelVersions: {
+      1: {
+        modelChange: {
+          type: 'expansion',
+        },
+      },
+    },
+  });
+
+  const typeV2 = createType({
+    name: 'sample_type',
+    mappings: {
+      properties: {
+        text: { type: 'text' },
+        keyword: { type: 'keyword' },
+        newField1: { type: 'text' },
+      },
+    },
+    switchToModelVersionAt: '8.0.0',
+    modelVersions: {
+      1: {
+        modelChange: {
+          type: 'expansion',
+        },
+      },
+      2: {
+        modelChange: {
+          type: 'expansion',
+          addedMappings: {
+            newField1: { type: 'text' },
+          },
+        },
+      },
+    },
+  });
+
+  const createBaseline = async () => {
+    const { runMigrations, savedObjectsRepository } = await getKibanaMigratorTestKit({
+      ...getBaseMigratorParams(),
+      types: [typeV2],
+    });
+    await runMigrations();
+
+    const sampleAObjs = range(5).map<SavedObjectsBulkCreateObject>((number) => ({
+      id: `doc-${number}`,
+      type: 'sample_type',
+      attributes: {
+        text: `text ${number}`,
+        keyword: `kw ${number}`,
+        newField1: `was added in v2`,
+      },
+    }));
+
+    await savedObjectsRepository.bulkCreate(sampleAObjs);
+  };
+
+  it('migrates the documents', async () => {
+    await createBaseline();
+
+    const { runMigrations, client, savedObjectsRepository } = await getKibanaMigratorTestKit({
+      ...getBaseMigratorParams(),
+      logFilePath,
+      types: [typeV1],
+    });
+
+    await runMigrations();
+
+    const indices = await client.indices.get({ index: '.kibana*' });
+    expect(Object.keys(indices)).toEqual(['.kibana_1']);
+
+    const index = indices['.kibana_1'];
+    const mappings = index.mappings ?? {};
+    const mappingMeta = mappings._meta ?? {};
+
+    // check that mappings are still matching v2
+    expect(mappings.properties).toEqual(
+      expect.objectContaining({
+        sample_type: typeV2.mappings,
+      })
+    );
+
+    // check that the mappingVersions are still v2
+    expect(mappingMeta.mappingVersions).toEqual({
+      sample_type: '10.2.0',
```

Comment:
Ignorant question: How / where is this _mappingVersions_ `10.2.0` defined?

Reply:
Model version 2 = virtual version `10.2.0`.